src/main/scala/uk/gov/hmrc/ct/box/Validators.scala

```diff
@@ -16,50 +16,50 @@
 
 package uk.gov.hmrc.ct.box
 
-import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
+import uk.gov.hmrc.ct.box.retriever.BoxRetriever
 import uk.gov.hmrc.ct.domain.ValidationConstants._
 
-trait Validators {
+trait Validators[B <: BoxRetriever] {
 
   protected val boxId = getClass.getSimpleName
 
-  protected def And(predicates: ((ComputationsBoxRetriever) => Boolean)*)(boxRetriever: ComputationsBoxRetriever): Boolean = {
+  protected def And(predicates: ((B) => Boolean)*)(boxRetriever: B): Boolean = {
```

Comment:
would have the boxRetriever as implicit and also

def validate(boxRetriever: T): Set[CtValidation] in trait ValidatableBox[T <: BoxRetriever]
as 
def validate(implicit boxRetriever: T): Set[CtValidation] 

lots of boilerplate code passing the box retriever around